### ISSUE TITLE
Contact appeared in contacts list twice

### ISSUE THREAD

**Issue by o---** [UWPX/UWPX-Client#145]
## Bevor you create a new issue:

```
- [X] I searched for similar issues and did not find one
- [X] I'm using the latest version available in the [Windows Store](https://www.microsoft.com/store/apps/9NW16X9JB5WV)
```

## I'm submitting a...:

 - Bug report (I searched for similar issues and did not find one)
 

## Current behavior:
I have received an unencrypted message from a contact which I did not have on the left side in the contact list. 
However the contact appeared not once, but twice. 
![obrázok](https://user-images.githubusercontent.com/5457048/104643297-80312480-56ac-11eb-973e-ea8fb04152b3.png)

## Expected behavior:
The contact appears only once. 

## Environment:

```
App Version(s): 
v.0.30.0.0 

Windows 10 Version Number: <!-- https://en.wikipedia.org/wiki/Windows_10_version_history -->
- [X] 1809
- [ ] 1809
- [ ] 1803
- [ ] 1709
- [ ] 1703
- [ ] 1607
- [ ] 1511
- [ ] 1507
- [ ] Insider Build (build number: )
- [ ] Misc:

Device form factor:
- [X] Desktop
- [ ] Mobile
- [ ] Xbox
- [ ] Surface Hub

Where did you got the APP from?
- [X] [Windows Store](https://www.microsoft.com/store/apps/9NW16X9JB5WV)
- [ ] Self-build, using a provided release source
- [ ] Self-build, repo cloned at [dd.mm.yyy] <!-- When did you clone the repo! -->
- [ ] Misc, got it from... <!-- Please tell us your source! -->

```

**Comment by COM8** [comment]
Thanks for reporting this.
I'm sad to say, but I'm aware of this. I also noticed this behavior, especially when a lot of new chats get added at once like on initial connect with a new account.

This is probably a synchronization issue since the chat list it self is maintained by an other thread, than the one, that runs the UI.

I tried to fix this a bit of time ago, but I was not able to find a reliable way to reproduce this.
Looks like I have to have a look at this again.

As always I really appreciate the time you take to repot those issues. This really helps!
